### PR TITLE
Delete release notes branch on action fail

### LINKED ISSUE #1919: Address TODO in release_notes.yml
### Description
`release_notes.yml` includes the TODO: `Add a job to delete branch on PR failure.`

In this action, a branch is created to push release note changes to. If the action fails after this branch is created, we cannot successfully re-run the action without seeing a branch name collision. We must first delete the branch and re-run the job. A job to automatically delete the branch on action failure would solve this. 

Thoughts: Check out the newrelic-ruby-agent repository to access the recently created branch_name through `generate_release_notes.rb`. Then carry that name (as an env var) to the next step of checking out newrelic/docs-website to delete the branch. If it is possible to carry the branch_name env var between yml jobs, we could also use that. 

Here are some initial code thoughts (not working):

```
  delete_branch_on_fail:
    name: Delete branch on fail
    needs: [send-pull-requests]
    runs-on: ubuntu-22.04
    if: always()
    steps:

# Checkout newrelic-ruby-agent
      - name: Checkout repository
        uses: actions/checkout@v3
        with:
          repository: newrelic/newrelic-ruby-agent
      - name: Get branch name
        run: echo "branch_name=$(ruby -r /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/.github/workflows/scripts/generate_release_notes.rb -e GenerateReleaseNotes.new.branch_name)" >> $GITHUB_ENV

# Check out docs-website and delete branch
      - name: Checkout repository
        uses: actions/checkout@v3
        with:
          repository: newrelic/docs-website
          token: ${{ secrets.NEWRELIC_RUBY_AGENT_BOT_TOKEN }}
      - name: Build delete command
        run: echo "delete_branch=git push origin --delete --force ${{env.branch_name}}" >> $GITHUB_ENV
      - name: Delete branch
        uses: technote-space/workflow-conclusion-action@v3
      - run: ${{env.delete_branch}}
        if: ${{ env.WORKFLOW_CONCLUSION == 'failure' && github.event_name != 'workflow_dispatch' }}
```

### PR DESCRIPTION
**Pull request by hannahramadan**
This PR adds a new job to the `release_notes` action to delete a newly created branch upon job failure. Without this step, we couldn't re-run the action before manually deleting the branch.

A test run demonstrates a branch successfully created in the job `send-pull-request` that fails at the `make pull request step`. In the `Delete branch on fail job` we are successfully deleting the branch: https://github.com/newrelic/newrelic-ruby-agent/actions/runs/5294926446

Here is the job not running if successful: https://github.com/newrelic/newrelic-ruby-agent/actions/runs/5294997587

Resolves #1919

### CODE REVIEW

**Review by kaylareopelle — APPROVED** [review]
This looks great! Nice job!

File: .github/workflows/release_notes.yml [hunk 1]
```diff
@@ -50,4 +50,28 @@ jobs:
           pr_body: "This is an automated PR generated when the Ruby agent is released. Please merge as soon as possible."
           destination_repository: "newrelic/docs-website"
 
-# TODO: Add a job to delete branch on PR failure. 
+  delete_branch_on_fail:
+    name: Delete branch on fail
+    needs: [send-pull-requests]
+    runs-on: ubuntu-22.04
+    if: always()
+    steps:
```
Comment (fallwith) [inline .github/workflows/release_notes.yml:58]: (as discussed offline) Ideally we don't want to run _any_ of these new 5 steps unless the workflow conclusion is "failure", but currently we'll perform steps 1-4 in ALL cases and then only guard the 5th step. This isn't a terribly big deal given the infrequency of the release notes workflow being called.

File: .github/workflows/release_notes.yml [hunk 2]
```diff
@@ -50,4 +50,28 @@ jobs:
...
+          repository: newrelic/docs-website
+          token: ${{ secrets.NEWRELIC_RUBY_AGENT_BOT_TOKEN }}
+
+      - name: Build delete command
+        run: echo "delete_file=git push origin --delete ${{env.branch_name}} --force" >> $GITHUB_ENV
+
+      - name: Delete branch on fail
+        uses: technote-space/workflow-conclusion-action@v3
+      - run: ${{env.delete_file}}
+        if: ${{ env.WORKFLOW_CONCLUSION == 'failure' && github.event_name != 'workflow_dispatch' }}
```
Comment (kaylareopelle) [inline .github/workflows/release_notes.yml:84]: Do you want to keep the `workflow_dispatch` exception here? Is it intended for testing?

File: .github/workflows/release_notes.yml (hunk 1 above)
Comment (hannahramadan) [reply .github/workflows/release_notes.yml:57]: This has been fixed by adding `if: failure()`

**Review by kaylareopelle — APPROVED** [review]

**Review by fallwith — APPROVED** [review]

**Pull request merged**